formats/gallery/Gallery.php

```diff
@@ -316,7 +316,18 @@ protected function addImageToGallery( ImageGallery &$ig, Title $imgTitle, $imgCa
 		}
 		// Use image alt as helper for either text
 		$imgAlt =  $this->params['redirects'] === '' ? $imgCaption : $imgRedirect !== '' ? $imgRedirect : '' ;
-		$ig->add( $imgTitle, $imgCaption, $imgAlt );
+
+		if ($this->params['lightgallery']) {
+			$file = wfFindFile( $imgTitle );
+			if ( $file && $file->exists() ) {
+				$url = $file->getUrl();
+				$ig->add( $imgTitle, $imgCaption, $imgAlt, $url );
+			} else { // Not sure what to do here
+				// $ig->add( $imgTitle, $imgCaption, $imgAlt );
+			}
+		} else {
+			$ig->add( $imgTitle, $imgCaption, $imgAlt );
+		}
```

Comment:
This function contains a lot of if-else... perhaps something can be split out of it?